<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yuxin Zou" />


<title>Expression 18 Mash - reprocessalltime - EE - joint mean</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mash_John_2015</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Expression 18 Mash - reprocessalltime - EE - joint mean</h1>
<h4 class="author">Yuxin Zou</h4>
<h4 class="date">2021-2-25</h4>

</div>


<pre class="r"><code>library(limma); library(mashr); library(plyr);</code></pre>
<pre><code>## Loading required package: ashr</code></pre>
<pre class="r"><code>library(flashr); library(colorRamps); library(corrplot)</code></pre>
<pre><code>## corrplot 0.84 loaded</code></pre>
<pre class="r"><code>library(ggplot2); library(ggpubr)</code></pre>
<pre><code>## 
## Attaching package: &#39;ggpubr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:plyr&#39;:
## 
##     mutate</code></pre>
<p>Implement <code>mash</code> on John’s data at time point 18.</p>
<pre class="r"><code>data = readRDS(&#39;../output/Expression_reprocessalltime18_joint.rds&#39;)
beta_class = data$beta
sebeta_class = data$sebeta
colnames(beta_class) = colnames(sebeta_class) = c(&#39;none&#39;, &#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;)

mash.data = mash_set_data(beta_class, sebeta_class)
mash.data.L = mash_update_data(mash.data, ref=&#39;none&#39;)</code></pre>
<p>Top genes:</p>
<pre class="r"><code># find strong genes
m.1by1 = mash_1by1(mash.data.L, alpha=0)
strong = get_significant_results(m.1by1)

data.strong = mash_set_data(Bhat = beta_class[strong,], Shat = sebeta_class[strong,])
data.strong.L = mash_update_data(data.strong, ref = &#39;none&#39;)</code></pre>
<p>There are 5570 strong genes for at least one bacteria using <code>ash</code> condition-by-condition.</p>
<div id="estimate-covariance-structures" class="section level2">
<h2>Estimate covariance structures</h2>
<p>Data Driven:</p>
<p>Flash:</p>
<pre class="r"><code>flash.data = flash_set_data(data.strong.L$Bhat)
fmodel = flash(flash.data, greedy = TRUE, backfit = TRUE)</code></pre>
<pre class="r"><code>factors = flash_get_ldf(fmodel)$f
row.names(factors) = colnames(beta_class)[2:9]
pve.order = order(fmodel$pve, decreasing = TRUE)

par(mfrow=c(2,3))
for(i in pve.order){
  barplot(factors[,i], main=paste0(&#39;Factor &#39;,i, &#39; pve= &#39;, round(fmodel$pve[i],3)), las=2, cex.names = 0.7)
}</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/flash%20plot-1.png" width="672" /></p>
<pre class="r"><code>par(mfrow=c(1,1))</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/flash%20plot-2.png" width="672" /></p>
<pre class="r"><code>Factors = fmodel$ldf$f
Flash_res = flash_get_fitted_values(fmodel)
U.Flash = c(mashr:::cov_from_factors(t(as.matrix(Factors)), &quot;Flash&quot;), 
            list(&quot;tFlash&quot; = t(Flash_res) %*% Flash_res / nrow(Flash_res)))</code></pre>
<p>PCA:</p>
<pre class="r"><code>U.pca = cov_pca(data.strong.L, 3)</code></pre>
<p>Canonical</p>
<pre class="r"><code>U.c = cov_canonical(data.strong.L)</code></pre>
<p>Extreme Deconvolution</p>
<pre class="r"><code>X.center = apply(data.strong.L$Bhat, 2, function(x) x - mean(x))
U.dd = c(U.pca, U.Flash, list(&#39;XX&#39; = t(X.center) %*% X.center / nrow(X.center)))

saveRDS(cov_ed(data.strong.L, U.dd), &#39;../output/Expression18_reprocessalltime_EE_jointmean_ED.rds&#39;)</code></pre>
</div>
<div id="mash-model" class="section level2">
<h2>Mash model</h2>
<pre class="r"><code>U.ed = readRDS(&#39;../output/Expression18_reprocessalltime_EE_jointmean_ED.rds&#39;)</code></pre>
<pre class="r"><code>U.dd.diag = c(U.ed[c(&#39;ED_tPCA&#39;, &#39;ED_tFlash&#39;, &#39;ED_XX&#39;)], lapply(U.ed[c(&#39;ED_PCA_1&#39;,&#39;ED_PCA_2&#39;,&#39;ED_PCA_3&#39;, paste0(&#39;ED_Flash_&#39;,1:7))], function(U) U + 0.01*diag(8)))
mash.model = mash(mash.data.L, c(U.c, U.dd.diag))
saveRDS(mash.model, &#39;../output/Expression18_reprocessalltime_EE_jointmean_mash_model.rds&#39;) </code></pre>
</div>
<div id="result" class="section level2">
<h2>Result</h2>
<pre class="r"><code>mash.model = readRDS(&#39;../output/Expression18_reprocessalltime_EE_jointmean_mash_model.rds&#39;)</code></pre>
<p>The log-likelihood of fit is</p>
<pre class="r"><code>get_loglik(mash.model)</code></pre>
<pre><code>## [1] -28785.29</code></pre>
<p>Here is a plot of weights learned:</p>
<pre class="r"><code>options(repr.plot.width=12, repr.plot.height=4)
barplot(get_estimated_pi(mash.model), las = 2, cex.names = 0.7)</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>There are 6377 differentially expressed genes.</p>
<p>Check tFlash covariance matrix</p>
<pre class="r"><code>x           &lt;- mash.model$fitted_g$Ulist[[&quot;ED_tFlash&quot;]]
colnames(x) &lt;- colnames(beta_class)[2:9]
rownames(x) &lt;- colnames(x)
corrplot(x, method=&#39;color&#39;, cl.lim=c(0,1), type=&#39;upper&#39;, addCoef.col = &quot;black&quot;, tl.col=&quot;black&quot;, tl.srt=45, tl.cex = 0.7)</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/tFlash-1.png" width="672" /></p>
<pre class="r"><code>layout(matrix(c(1,2,3,4), 2, 2, byrow=TRUE))
svd.out = svd(mash.model$fitted_g$Ulist[[&quot;ED_tFlash&quot;]])
v = svd.out$v
colnames(v) = colnames(beta_class)[-1]
rownames(v) = colnames(v)
options(repr.plot.width=10, repr.plot.height=5)
for (j in 1:4)
  barplot(v[,j]/v[,j][which.max(abs(v[,j]))], cex.names = 0.7,
          las = 2, main = paste0(&quot;EigenVector &quot;, j, &quot; for ED_tFlash&quot;))</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/tFLASH%20pca-1.png" width="672" /></p>
<pre class="r"><code>png(&#39;../output/Expression18_EE_mash_model_tFlash_v1.png&#39;)
svd.out = svd(mash.model$fitted_g$Ulist[[&quot;ED_tFlash&quot;]])
v = svd.out$v
colnames(v) = colnames(beta_class)[-1]
rownames(v) = colnames(v)
barplot(v[,1]/v[,1][which.max(abs(v[,1]))], cex.names = 0.7, las = 2, 
        main = paste0(&quot;EigenVector 1, pve=&quot;, round(svd.out$d[1]/sum(svd.out$d), 2)))
dev.off()</code></pre>
<p>Check tPCA covariance matrix</p>
<pre class="r"><code>x           &lt;- mash.model$fitted_g$Ulist[[&quot;ED_tPCA&quot;]]
colnames(x) &lt;- colnames(beta_class)[2:9]
rownames(x) &lt;- colnames(x)
corrplot(x, method=&#39;color&#39;, cl.lim=c(0,1), type=&#39;upper&#39;, addCoef.col = &quot;black&quot;, tl.col=&quot;black&quot;, tl.srt=45, tl.cex = 0.7)</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/tPCA-1.png" width="672" /></p>
<pre class="r"><code>layout(matrix(c(1,2,3,4), 2, 2, byrow=TRUE))
svd.out = svd(mash.model$fitted_g$Ulist[[&quot;ED_tPCA&quot;]])
v = svd.out$v
colnames(v) = colnames(beta_class)[-1]
rownames(v) = colnames(v)
options(repr.plot.width=10, repr.plot.height=5)
for (j in 1:4)
  barplot(v[,j]/v[,j][which.max(abs(v[,j]))], cex.names = 0.7,
          las = 2, main = paste0(&quot;EigenVector &quot;, j, &quot; for ED_tPCA&quot;))</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/tPCA%20pca-1.png" width="672" /></p>
<p>Check pairwise sharing by magnitude and sign:</p>
<pre class="r"><code>x = get_pairwise_sharing(mash.model)
x[x &gt; 1]    &lt;- 1
x[x &lt; -1]   &lt;- -1
colnames(x) &lt;- colnames(beta_class)[-1]
rownames(x) &lt;- colnames(x)
corrplot.mixed(x, tl.pos=&quot;d&quot;,upper=&#39;color&#39;, cl.lim=c(0,1), upper.col=colorRampPalette(rev(c(&quot;#D73027&quot;,&quot;#FC8D59&quot;,&quot;#FEE090&quot;,&quot;#FFFFBF&quot;,
                               &quot;#E0F3F8&quot;,&quot;#91BFDB&quot;,&quot;#4575B4&quot;)))(40),
               tl.cex=1.2)</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/sharingMagSign-1.png" width="672" /></p>
<p>Check pairwise sharing by magnitude:</p>
<pre class="r"><code>x = get_pairwise_sharing(mash.model, FUN = abs)
x[x &gt; 1]    &lt;- 1
x[x &lt; -1]   &lt;- -1
colnames(x) &lt;- colnames(beta_class)[-1]
rownames(x) &lt;- colnames(x)
corrplot.mixed(x, tl.pos=&quot;d&quot;,upper=&#39;color&#39;, cl.lim=c(0,1), upper.col=colorRampPalette(rev(c(&quot;#D73027&quot;,&quot;#FC8D59&quot;,&quot;#FEE090&quot;,&quot;#FFFFBF&quot;,
                               &quot;#E0F3F8&quot;,&quot;#91BFDB&quot;,&quot;#4575B4&quot;)))(40),
               tl.cex=1.2)</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/shareMag-1.png" width="672" /></p>
<p>Check pairwise sharing by sign:</p>
<pre class="r"><code>x = get_pairwise_sharing(mash.model, factor=0)
x[x &gt; 1]    &lt;- 1
x[x &lt; -1]   &lt;- -1
colnames(x) &lt;- colnames(beta_class)[-1]
rownames(x) &lt;- colnames(x)
corrplot.mixed(x, tl.pos=&quot;d&quot;,upper=&#39;color&#39;, cl.lim=c(0,1), upper.col=colorRampPalette(rev(c(&quot;#D73027&quot;,&quot;#FC8D59&quot;,&quot;#FEE090&quot;,&quot;#FFFFBF&quot;,
                               &quot;#E0F3F8&quot;,&quot;#91BFDB&quot;,&quot;#4575B4&quot;)))(40),
               tl.cex=1.2)</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/shareSign-1.png" width="672" /></p>
</div>
<div id="genes" class="section level2">
<h2>Genes</h2>
<p>Yers-Salm cluster in Cormotif, significant in mash.</p>
<pre class="r"><code># before
gene1177 = data.frame(mash.data.L$Bhat[1177,])
colnames(gene1177) = &#39;Deviations&#39;
gene1177$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene1177$se = mash.data.L$Shat[1177,]
gene1177$Deviations[gene1177$se == 0] = NA
# after
gene1177.post = data.frame(mash.model$result$PosteriorMean[1177,])
colnames(gene1177.post) = &#39;Deviations&#39;
gene1177.post$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene1177.post$se = mash.model$result$PosteriorSD[1177,]

p.orig = ggplot(gene1177, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + ylim(-1,3) +
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) + 
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000079335 original&#39; )

p.post = ggplot(gene1177.post, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + ylim(-1,3) +
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) + 
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000079335 mash&#39;)

ggarrange(p.orig, p.post, ncol=2, nrow=1, heights = c(5, 5), widths = c(5,5))</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/YSgene1-1.png" width="960" /></p>
<p>Yers-Salm cluster in Cormotif, significant in mash.</p>
<pre class="r"><code># before
gene7637 = data.frame(mash.data.L$Bhat[7637,])
colnames(gene7637) = &#39;Deviations&#39;
gene7637$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene7637$se = mash.data.L$Shat[7637,]
gene7637$Deviations[gene7637$se == 0] = NA
# after
gene7637.post = data.frame(mash.model$result$PosteriorMean[7637,])
colnames(gene7637.post) = &#39;Deviations&#39;
gene7637.post$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene7637.post$se = mash.model$result$PosteriorSD[7637,]

p.orig = ggplot(gene7637, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + ylim(-8,3) +
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) + 
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000158488 original&#39; )

p.post = ggplot(gene7637.post, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + ylim(-8, 3) +
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) + 
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000158488 mash&#39;)

ggarrange(p.orig, p.post, ncol=2, nrow=1, heights = c(5, 5), widths = c(5,5))</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/YSgene2-1.png" width="960" /></p>
<p>All in Cormofit, non-significant in mash.</p>
<pre class="r"><code># before
gene4 = data.frame(mash.data.L$Bhat[4,])
colnames(gene4) = &#39;Deviations&#39;
gene4$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene4$se = mash.data.L$Shat[4,]
gene4$Deviations[gene4$se == 0] = NA
# after
gene4.post = data.frame(mash.model$result$PosteriorMean[4,])
colnames(gene4.post) = &#39;Deviations&#39;
gene4.post$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene4.post$se = mash.model$result$PosteriorSD[4,]

p.orig = ggplot(gene4, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + ylim(-6,4) +
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) +
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000000938 original&#39; )

p.post = ggplot(gene4.post, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + ylim(-6, 4) +
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) + 
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000000938 mash&#39;)

ggarrange(p.orig, p.post, ncol=2, nrow=1, heights = c(5, 5), widths = c(5,5))</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/allgene1-1.png" width="960" /></p>
<p>All in Cormofit, significant in mash.</p>
<pre class="r"><code># before
gene11445 = data.frame(mash.data.L$Bhat[11445,])
colnames(gene11445) = &#39;Deviations&#39;
gene11445$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene11445$se = mash.data.L$Shat[11445,]
gene11445$Deviations[gene11445$se == 0] = NA
# after
gene11445.post = data.frame(mash.model$result$PosteriorMean[11445,])
colnames(gene11445.post) = &#39;Deviations&#39;
gene11445.post$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene11445.post$se = mash.model$result$PosteriorSD[11445,]

p.orig = ggplot(gene11445, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + ylim(-1,4) +
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) +
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000196639 original&#39; )

p.post = ggplot(gene11445.post, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + ylim(-1, 4) +
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) + 
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000196639 mash&#39;)

ggarrange(p.orig, p.post, ncol=2, nrow=1, heights = c(5, 5), widths = c(5,5))</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/allgene2-1.png" width="960" /></p>
<p>All in Cormofit, significant in mash.</p>
<pre class="r"><code># before
gene2481 = data.frame(mash.data.L$Bhat[2481,])
colnames(gene2481) = &#39;Deviations&#39;
gene2481$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene2481$se = mash.data.L$Shat[2481,]
gene2481$Deviations[gene2481$se == 0] = NA
# after
gene2481.post = data.frame(mash.model$result$PosteriorMean[2481,])
colnames(gene2481.post) = &#39;Deviations&#39;
gene2481.post$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene2481.post$se = mash.model$result$PosteriorSD[2481,]

p.orig = ggplot(gene2481, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + 
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) +
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000105497 original&#39; )

p.post = ggplot(gene2481.post, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + 
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) + 
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000105497 mash&#39;)

ggarrange(p.orig, p.post, ncol=2, nrow=1, heights = c(5, 5), widths = c(5,5))</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/allgene3-1.png" width="960" /></p>
<p>Non-DE in Cormofit, significant in mash.</p>
<pre class="r"><code># before
gene19 = data.frame(mash.data.L$Bhat[19,])
colnames(gene19) = &#39;Deviations&#39;
gene19$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene19$se = mash.data.L$Shat[19,]
gene19$Deviations[gene19$se == 0] = NA
# after
gene19.post = data.frame(mash.model$result$PosteriorMean[19,])
colnames(gene19.post) = &#39;Deviations&#39;
gene19.post$Group = factor(c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;), levels = c(&#39;Rv&#39;, &#39;Rvplus&#39;, &#39;GC&#39;, &#39;BCG&#39;, &#39;Smeg&#39;, &#39;Yers&#39;, &#39;Salm&#39;, &#39;Staph&#39;))
gene19.post$se = mash.model$result$PosteriorSD[19,]

p.orig = ggplot(gene19, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + 
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) +
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000002549 original&#39; )

p.post = ggplot(gene19.post, aes(y = Deviations, x = Group, color=Group)) + 
  geom_point() + 
  geom_errorbar(aes(ymin=Deviations-1.96*se, ymax=Deviations+1.96*se), width=0.4) + 
  geom_hline(yintercept = 0, linetype=&quot;dotted&quot;, color = &quot;grey&quot;, size=1) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(),axis.line = element_line(colour = &quot;black&quot;), legend.position = &quot;none&quot;, plot.title = element_text(size = 10), axis.text=element_text(size=10)) + coord_flip() + ggtitle(&#39;ENSG00000002549 mash&#39;)

ggarrange(p.orig, p.post, ncol=2, nrow=1, heights = c(5, 5), widths = c(5,5))</code></pre>
<p><img src="Expression18_EE_reprocess_joint_files/figure-html/allgene4-1.png" width="960" /></p>
</div>
<div id="session-information" class="section level1">
<h1>Session information</h1>
<!-- Insert the session information into the document -->
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
