<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yuxin Zou" />


<title>Expression simulation under null</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mash_John_2015</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Expression simulation under null</h1>
<h4 class="author">Yuxin Zou</h4>
<h4 class="date">2021-2-19</h4>

</div>


<pre class="r"><code>library(limma)
library(ashr)
library(mashr)</code></pre>
<p>The data includes the gene expression data (table_s1) of 12,728 genes from 54 samples infected by 8 different types of bacteria and one non-infected control, measured at 18 hours post-infection.</p>
<div id="same-permutation-for-all-gene" class="section level2">
<h2>Same permutation for all gene</h2>
<p>We randomly permute the samples with respect to the condition labels so that there no longer exists association between gene expression levels and conditions.</p>
<pre class="r"><code>dat = readRDS(&#39;../output/Expression_data.rds&#39;)
set.seed(123)

perm = sample(1:54, 54)
d1 = dat$data[,perm]
colnames(d1) = colnames(dat$data)
  
fit1=lmFit(object = d1, dat$design_diff)
fit1&lt;-eBayes(fit1)
b1 = fit1$coefficients[,-1]
s1 &lt;- fit1$stdev.unscaled[,-1] * sqrt(fit1$s2.post)
t1 &lt;- fit1$t[,-1]
df1 &lt;- fit1$df.total

fit2=lmFit(object = d1, dat$design_mean)
fit2&lt;-eBayes(fit2)
b2 = fit2$coefficients
s2 &lt;- fit2$stdev.unscaled * sqrt(fit2$s2.post)
t2 &lt;- fit2$t
df2 &lt;- fit2$df.total</code></pre>
<p>Fit ash for each condition:</p>
<pre class="r"><code>m1.data = mash_set_data(b1, s1)
m1.1by1 = mash_1by1(m1.data)
fpash = colSums(m1.1by1$result$lfsr &lt; 0.05)</code></pre>
<p>The false positive rates using ash:</p>
<pre class="r"><code>fpash/nrow(d1)</code></pre>
<pre><code>##           Rv       Rvplus           GC          BCG         Smeg         Yers 
## 0.0014142049 0.0000000000 0.0000000000 0.0000000000 0.0051068510 0.0001571339 
##         Salm        Staph 
## 0.0006285355 0.0000000000</code></pre>
<p>Fit mash on DE data:</p>
<pre class="r"><code>strong1 = get_significant_results(m1.1by1, 0.05)
U1.pca = cov_pca(m1.data, 5, subset=strong1)
U1.ed = cov_ed(m1.data, U1.pca, subset=strong1)
U1.c = cov_canonical(m1.data)
m1.fit.diff = mash(m1.data, c(U1.c,U1.ed))</code></pre>
<pre><code>##  - Computing 12728 x 419 likelihood matrix.
##  - Likelihood calculations took 7.72 seconds.
##  - Fitting model with 419 mixture components.
##  - Model fitting took 29.44 seconds.
##  - Computing posterior matrices.
##  - Computation allocated took 11.48 seconds.</code></pre>
<pre class="r"><code>fpmash.diff = length(get_significant_results(m1.fit.diff))</code></pre>
<p>The FPR is 0.3399591. We check estimated weights:</p>
<pre class="r"><code>barplot(get_estimated_pi(m1.fit.diff), las=2, cex.names = 0.7)</code></pre>
<p><img src="Expression18_jointdiff_simulatenull_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>The ED priors already adjusted with a small diagonal elements. We add a small number on the diagonal of cannonical priors:</p>
<pre class="r"><code>U1.c.diag = lapply(U1.c, function(U) U + 0.01 * diag(8))
m1.fit.diff.diag = mash(m1.data, c(U1.c.diag, U1.ed))</code></pre>
<pre><code>##  - Computing 12728 x 419 likelihood matrix.
##  - Likelihood calculations took 7.86 seconds.
##  - Fitting model with 419 mixture components.
##  - Model fitting took 33.16 seconds.
##  - Computing posterior matrices.
##  - Computation allocated took 8.48 seconds.</code></pre>
<pre class="r"><code>fpmash.diff.diag = length(get_significant_results(m1.fit.diff.diag))</code></pre>
<p>The FPR is 0.3370522. We check estimated weights:</p>
<pre class="r"><code>barplot(get_estimated_pi(m1.fit.diff.diag), las=2, cex.names = 0.7)</code></pre>
<p><img src="Expression18_jointdiff_simulatenull_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>Fit mash common baseline:</p>
<pre class="r"><code>m2.data = mash_set_data(b2, s2)
m2.data.L = mash_update_data(m2.data, ref=&#39;none&#39;)
m2.1by1 = mash_1by1(m2.data.L)
strong2 = get_significant_results(m2.1by1, 0.05)
U2.pca = cov_pca(m2.data.L, 5, subset=strong2)
U2.ed = cov_ed(m2.data.L, U2.pca, subset=strong2)
U2.c = cov_canonical(m2.data.L)

m2.fit.mean = mash(m2.data.L, c(U2.c,U2.ed))</code></pre>
<pre><code>##  - Computing 12728 x 419 likelihood matrix.
##  - Likelihood calculations took 8.18 seconds.
##  - Fitting model with 419 mixture components.
##  - Model fitting took 34.68 seconds.
##  - Computing posterior matrices.
##  - Computation allocated took 7.37 seconds.</code></pre>
<pre class="r"><code>fpmash.mean = length(get_significant_results(m2.fit.mean))</code></pre>
<p>The FPR is 0.1851037. We check estimated weights:</p>
<pre class="r"><code>barplot(get_estimated_pi(m2.fit.mean), las=2, cex.names = 0.7)</code></pre>
<p><img src="Expression18_jointdiff_simulatenull_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p>Add a small number on the diagonal of priors:</p>
<pre class="r"><code>U2.c.diag = lapply(U2.c, function(U) U + 0.01 * diag(8))
m2.fit.mean.diag = mash(m2.data.L, c(U2.c.diag, U2.ed))</code></pre>
<pre><code>##  - Computing 12728 x 419 likelihood matrix.
##  - Likelihood calculations took 7.81 seconds.
##  - Fitting model with 419 mixture components.
##  - Model fitting took 32.12 seconds.
##  - Computing posterior matrices.
##  - Computation allocated took 7.08 seconds.</code></pre>
<pre class="r"><code>fpmash.mean.diag = length(get_significant_results(m2.fit.mean.diag))</code></pre>
<p>The FPR is 0.1704117. We check estimated weights:</p>
<pre class="r"><code>barplot(get_estimated_pi(m2.fit.mean.diag), las=2, cex.names = 0.7)</code></pre>
<p><img src="Expression18_jointdiff_simulatenull_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>If we exclude data-driven priors,</p>
<pre class="r"><code>m2.fit.mean.c = mash(m2.data.L, U2.c)</code></pre>
<pre><code>##  - Computing 12728 x 287 likelihood matrix.
##  - Likelihood calculations took 5.34 seconds.
##  - Fitting model with 287 mixture components.
##  - Model fitting took 18.96 seconds.
##  - Computing posterior matrices.
##  - Computation allocated took 9.15 seconds.</code></pre>
<pre class="r"><code>fpmash.mean.c = length(get_significant_results(m2.fit.mean.c))</code></pre>
<p>The FPR is 0.1851037. We check estimated weights:</p>
<pre class="r"><code>barplot(get_estimated_pi(m2.fit.mean.c), las=2, cex.names = 0.7)</code></pre>
<p><img src="Expression18_jointdiff_simulatenull_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
</div>
<div id="different-permutation-for-each-gene" class="section level2">
<h2>Different permutation for each gene</h2>
<p>We use independent and different permutation to each gene.</p>
<pre class="r"><code>dat = readRDS(&#39;../output/Expression_data.rds&#39;)
set.seed(123)
d2 = dat$data
for(i in 1:nrow(dat$data)){
  perm = sample(1:54, 54)
  d2[i,] = d2[i,perm]
}

fit3=lmFit(object = d2, dat$design_diff)
fit3&lt;-eBayes(fit3)
b3 = fit3$coefficients[,-1]
s3 &lt;- fit3$stdev.unscaled[,-1] * sqrt(fit3$s2.post)
t3 &lt;- fit3$t[,-1]
df3 &lt;- fit3$df.total

fit4=lmFit(object = d2, dat$design_mean)
fit4&lt;-eBayes(fit4)
b4 = fit4$coefficients
s4 &lt;- fit4$stdev.unscaled * sqrt(fit4$s2.post)
t4 &lt;- fit4$t
df4 &lt;- fit4$df.total</code></pre>
<p>Fit ash for each condition:</p>
<pre class="r"><code>m3.data = mash_set_data(b3, s3)
m3.1by1 = mash_1by1(m3.data)
fpash.g = colSums(m3.1by1$result$lfsr &lt; 0.05)</code></pre>
<p>The false positive rates using ash:</p>
<pre class="r"><code>fpash.g/nrow(d2)</code></pre>
<pre><code>##           Rv       Rvplus           GC          BCG         Smeg         Yers 
## 0.000000e+00 0.000000e+00 0.000000e+00 7.856694e-05 0.000000e+00 0.000000e+00 
##         Salm        Staph 
## 0.000000e+00 0.000000e+00</code></pre>
<p>Fit mash on DE data:</p>
<pre class="r"><code>U3.c = cov_canonical(m3.data)
m3.fit.diff = mash(m3.data, U3.c)</code></pre>
<pre><code>##  - Computing 12728 x 300 likelihood matrix.
##  - Likelihood calculations took 6.24 seconds.
##  - Fitting model with 300 mixture components.
##  - Model fitting took 21.16 seconds.
##  - Computing posterior matrices.
##  - Computation allocated took 10.70 seconds.</code></pre>
<pre class="r"><code>fpmash.diff.g = length(get_significant_results(m3.fit.diff))</code></pre>
<p>The FPR is 0.2403363. We check estimated weights:</p>
<pre class="r"><code>barplot(get_estimated_pi(m3.fit.diff), las=2, cex.names = 0.7)</code></pre>
<p><img src="Expression18_jointdiff_simulatenull_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p>Fit mash common baseline:</p>
<pre class="r"><code>m4.data = mash_set_data(b4, s4)
m4.data.L = mash_update_data(m4.data, ref=&#39;none&#39;)
m4.1by1 = mash_1by1(m4.data.L)
U4.c = cov_canonical(m4.data.L)
m4.fit.mean = mash(m4.data.L, U4.c)</code></pre>
<pre><code>##  - Computing 12728 x 300 likelihood matrix.
##  - Likelihood calculations took 5.61 seconds.
##  - Fitting model with 300 mixture components.
##  - Model fitting took 19.96 seconds.
##  - Computing posterior matrices.
##  - Computation allocated took 24.00 seconds.</code></pre>
<pre class="r"><code>fpmash.mean.g = length(get_significant_results(m4.fit.mean))</code></pre>
<p>The FPR is 7.856693910^{-4}. We check estimated weights:</p>
<pre class="r"><code>barplot(get_estimated_pi(m4.fit.mean), las=2, cex.names = 0.7)</code></pre>
<p><img src="Expression18_jointdiff_simulatenull_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
