---
title: "Expression simulation under null"
author: "Yuxin Zou"
date: 2021-2-19
output: 
  html_document:
    code_folding: hide
---

```{r}
library(limma)
library(ashr)
```

The data includes the gene expression data (table_s1) of 12,728 genes from 54 samples infected by 8 different types of bacteria and one non-infected control, measured at 18 hours post-infection. We randomly permute the samples with respect to the condition labels so that there no longer exists association between gene expression levels and conditions. We use the same permutation to all the genes.

## ash on differential expression with original shat

```{r}
dat = readRDS('../output/Expression_data.rds')
set.seed(123)

dat_simu = list()
fp = matrix(NA, nrow=10, ncol=8)
for(i in 1:10){
  perm = sample(1:54, 54)
  
  d1 = dat$data[,perm]
  colnames(d1) = colnames(dat$data)
  dat_simu[[i]] = d1
  fit1=lmFit(object = d1, dat$design_diff)
  fit1<-eBayes(fit1)
  
  b1 = fit1$coefficients[,-1]
  s1 <- fit1$stdev.unscaled[,-1] * sqrt(fit1$s2.post)
  t1 <- fit1$t[,-1]
  df1 <- fit1$df.total
  
  for(j in 1:8){
    m1 = ash(b1[,j], s1[,j],mixcompdist="normal", alpha=0, lik=lik_normal())
    fp[i,j] = sum(m1$result$lfsr < 0.05)
  }
}
fp/nrow(dat$data)
```

## ash on differential expression with adjusted shat

```{r}
### take the function from mouthwash to calibrate shat
adjust_by_t <- function(betahat, sebetahat, df) {
  
  ## Convert t to z --------------------------------------
  zstats <- stats::qnorm(stats::pt(q = betahat / sebetahat, df = df))
  snew <- betahat / zstats
  
  return(snew)
}
```

```{r}
fpa = matrix(NA, nrow=10, ncol=8)
for(i in 1:10){
  
  d2 = dat_simu[[i]]
  fit2=lmFit(object = d2, dat$design_diff)
  fit2<-eBayes(fit2)
  
  b2 = fit2$coefficients[,-1]
  s2 <- fit2$stdev.unscaled[,-1] * sqrt(fit2$s2.post)
  t2 <- fit2$t[,-1]
  df2 <- fit2$df.total
  
  s2.adj = adjust_by_t(b2, s2, df=df2)
  
  for(j in 1:8){
    m2 = ash(b2[,j], s2.adj[,j],mixcompdist="normal", alpha=0, lik=lik_normal())
    fpa[i,j] = sum(m2$result$lfsr < 0.05)
  }
}
fpa/nrow(dat$data)
```

